credentials/tls/certprovider/pemfile/watcher.go

```diff
@@ -55,22 +56,41 @@ var (
 type Options struct {
 	// CertFile is the file that holds the identity certificate.
 	// Optional. If this is set, KeyFile must also be set.
-	CertFile string
+	CertFile string `json:"certificate_file,omitempty"`
 	// KeyFile is the file that holds identity private key.
 	// Optional. If this is set, CertFile must also be set.
-	KeyFile string
+	KeyFile string `json:"private_key_file,omitempty"`
 	// RootFile is the file that holds trusted root certificate(s).
 	// Optional.
-	RootFile string
+	RootFile string `json:"ca_certificate_file,omitempty"`
 	// SPIFFEBundleMapFile is the file that holds the spiffe bundle map.
 	// If a given provider configures both the RootFile and the
 	// SPIFFEBundleMapFile, the SPIFFEBundleMapFile will be preferred.
 	// Optional.
-	SPIFFEBundleMapFile string
+	SPIFFEBundleMapFile string `json:"spiffe_trust_bundle_map_file,omitempty"`
 	// RefreshDuration is the amount of time the plugin waits before checking
 	// for updates in the specified files.
 	// Optional. If not set, a default value (1 hour) will be used.
-	RefreshDuration time.Duration
+	RefreshDuration Duration `json:"refresh_interval,omitempty"`
+}
+
+// Duration wraps time.Duration to support Go-style duration strings in JSON (e.g., "10s", "5m").
+type Duration struct {
+	time.Duration
+}
+
+// UnmarshalJSON allows decoding a duration string like "200s" into time.Duration.
+func (d *Duration) UnmarshalJSON(b []byte) error {
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return fmt.Errorf("duration must be a string: %w", err)
+	}
+	dur, err := time.ParseDuration(s)
```

Comment:
The [gRFC](https://github.com/grpc/proposal/blob/master/A29-xds-tls-security.md#file_watcher-certificate-provider) mentions the following:
* `refresh_interval`: Specifies how frequently the plugin should read the files. The value must be in the JSON format described for a Duration protobuf message.

Based on this, using `time.ParseDuration` will deviate from the spec. Parsing a duration like "1m30s" would work with `time.ParseDuration` but fail when using `protojson.Unmarshal(cfg.RefreshInterval, &durationpb.Duration{})`. We could fix this by using protjosn and durationpb to parse in the here, but I'm not convinced addressing https://github.com/grpc/grpc-go/issues/8222 reduces complexity.

Reply:
> but fail when using protojson.Unmarshal(cfg.RefreshInterval, &durationpb.Duration{}). We could fix this by using protjosn and durationpb for parsing here,

@arjan-bal thx for comment, i have one question.
you mean it fails when using `protojson.Unmarshal(cfg.RefreshInterval, &durationpb.Duration{})`, and it could be fixed by using protojson and durationpb? i'm just bit confusing. i thought it's same meaning.

Reply:
`protojson.Unmarshal(cfg.RefreshInterval, &durationpb.Duration{}` will fail to parse a durations like "1m30s". This is the correct behaviour according to the gRFC. It may be that `time.ParseDuration` can parse a superset of what `durationpb` can (I haven't verified this), but we should not use `ParseDuration` to maintain cross language consistency.  If gRPC Go users start relying on the the use of `ParseDuration`, they'll see failures when the same config is used with other gRPC implementations.